### PR TITLE
Unity master new unitychanges profiles

### PR DESCRIPTION
**Pull request by UnityAlex**
Thought it would be a good idea to let you guys take a look at what was involved to get the unityjit-^OS^ and unityaot profiles created. I assume there are some definite things that we want removed as I somewhat blindly followed the net_4_x and winaot profiles when setting these up.

### CODE REVIEW

**Comment by joshpeterson** [comment]
Nothing here jumps out at me - this all looks sane. Thanks!

**Comment by joshpeterson** [comment]
I've pushed two commits to this branch to allow the profile stubber to work and to enable it in the build.

File: mcs/build/profiles/unityaot.make [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	-d:NET_2_1 \
+	-d:NET_3_5 \
+	-d:NET_4_0 \
+	-d:NET_4_5 \
+	-d:MONO \
+	-d:UNITY \
+	-d:UNITY_AOT \
+	-d:MOBILE,MOBILE_LEGACY \
+	-d:DISABLE_COM \
+	-d:WIN_PLATFORM \
```
Comment (joncham) [inline mcs/build/profiles/unityaot.make:25]: Fine for now, but if the aot profile will be cross platform the windows defines likely don't make sense.

File: mcs/class/corlib/unityaot_corlib.dll.sources [hunk 2]
```diff
@@ -0,0 +1 @@
+#include win32_build_corlib.dll.sources
```
Comment (joncham) [inline mcs/class/corlib/unityaot_corlib.dll.sources:1]: Again, win32 based. Fine for now but we'll need to address this.

**Comment by joncham** [comment]
Looks okay-ish. We'll need to figure out how to make unityaot be platform independent.

**Pull request merged**